symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftEmployeeAPI.php

```diff
@@ -0,0 +1,255 @@
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Pim\Api\Model\EmployeeDetailedModel;
+use OrangeHRM\Pim\Dto\EmployeeSearchFilterParams;
+use OrangeHRM\Pim\Service\Model\EmployeeModel;
+use OrangeHRM\Pim\Traits\Service\EmployeeServiceTrait;
+
+class WorkShiftEmployeeAPI extends EndPoint implements CollectionEndpoint
+{
+    use EmployeeServiceTrait;
+    use UserRoleManagerTrait;
+
+    public const FILTER_NAME = 'name';
+    public const FILTER_EMP_NUMBER = 'empNumber';
+    public const FILTER_NAME_OR_ID = 'nameOrId';
+    public const FILTER_EMPLOYEE_ID = 'employeeId';
+    public const FILTER_INCLUDE_EMPLOYEES = 'includeEmployees';
+    public const FILTER_EMP_STATUS_ID = 'empStatusId';
+    public const FILTER_JOB_TITLE_ID = 'jobTitleId';
+    public const FILTER_SUBUNIT_ID = 'subunitId';
+    public const FILTER_SUPERVISOR_EMP_NUMBERS = 'supervisorEmpNumbers';
+    public const FILTER_MODEL = 'model';
+    public const PARAM_RULE_EMPLOYEE_ID_MAX_LENGTH = 50;
+
+    public const PARAM_RULE_FILTER_NAME_MAX_LENGTH = 100;
+    public const PARAM_RULE_FILTER_NAME_OR_ID_MAX_LENGTH = 100;
+
+    public const MODEL_DEFAULT = 'default';
+    public const MODEL_DETAILED = 'detailed';
+    public const MODEL_MAP = [
+        self::MODEL_DEFAULT => EmployeeModel::class,
+        self::MODEL_DETAILED => EmployeeDetailedModel::class,
+    ];
+
+    public function getAll(): EndpointResult
+    {
+        $employeeParamHolder = new EmployeeSearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeParamHolder);
+
+        $empNumber = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_EMP_NUMBER
+        );
+        if (!is_null($empNumber)) {
+            $employeeParamHolder->setEmployeeNumbers([$empNumber]);
+        } else {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+            $employeeParamHolder->setEmployeeNumbers($accessibleEmpNumbers);
+        }
+
+        $employeeParamHolder->setIncludeEmployees(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_INCLUDE_EMPLOYEES
+            )
+        );
+        $employeeParamHolder->setName(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_NAME
+            )
+        );
```

Comment:
no need all filter fields, remove them here and in validation rules

Reply:
done